services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/SwapEndpointSpec.groovy

```diff
@@ -0,0 +1,887 @@
+        ]
+
+    }
+
+    def "Unable to swap endpoints for two flows when not enough bandwidth on ISL"() {
+        setup: "Create two flows with different source and the same destination switches"
+        def flow1SwitchPair = topologyHelper.getNeighboringSwitchPair()
+        def flow2SwitchPair = topologyHelper.getAllNeighboringSwitchPairs().find {
+            it.src != flow1SwitchPair.src && it.dst == flow1SwitchPair.dst
+        }
+        def flow1 = getFirstFlow(flow1SwitchPair, flow2SwitchPair)
+        def flow2 = getSecondFlow(flow1SwitchPair, flow2SwitchPair, flow1)
+
+        flowHelper.addFlow(flow1)
+        flowHelper.addFlow(flow2)
+
+        and: "Update the first flow so that it consumes all bandwidth on the link"
+        def flow1Path = PathHelper.convert(northbound.getFlowPath(flow1.id))
+        def flow1Isl = pathHelper.getInvolvedIsls(flow1Path)[0]
+        def flow1IslMaxBw = islUtils.getIslInfo(flow1Isl).get().maxBandwidth
+        northbound.updateFlow(flow1.id, flow1.tap { it.maximumBandwidth = flow1IslMaxBw })
+
+        and: "Break all alternative paths for the first flow"
+        def altPaths = flow1SwitchPair.paths.findAll { it != flow1Path }
+        List<PathNode> broughtDownPorts = []
+        altPaths.unique { it.first() }.each { path ->
+            def src = path.first()
+            broughtDownPorts.add(src)
+            northbound.portDown(src.switchId, src.portNo)
+        }
+        Wrappers.wait(WAIT_OFFSET) {
+            assert northbound.getAllLinks().findAll {
+                it.state == IslChangeType.FAILED
+            }.size() == broughtDownPorts.size() * 2
+        }
+
+        when: "Try to swap endpoints for two flows"
+        northbound.swapFlowEndpoint(
+                new SwapFlowPayload(flow1.id, flow1.source, changePropertyValue(flow1.destination, "portNumber",
+                        flow2.destination.portNumber)),
+                new SwapFlowPayload(flow2.id, changePropertyValue(flow2.source, "datapath", flow1.source.datapath),
+                        changePropertyValue(flow2.destination, "portNumber", flow1.destination.portNumber)))
+
+        then: "An error is received (400 code)"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 400
+        exc.responseBodyAsString.to(MessageError).errorMessage == "Can not swap endpoints for flows: " +
+                "Failed to find path with requested bandwidth=${flow2.maximumBandwidth}: " +
+                "Switch ${flow1SwitchPair.src.dpId} doesn't have links with enough bandwidth"
+
+        and: "Restore topology and delete flows"
+        broughtDownPorts.every { northbound.portUp(it.switchId, it.portNo) }
+        [flow1, flow2].each { flowHelper.deleteFlow(it.id) }
+        Wrappers.wait(discoveryInterval + WAIT_OFFSET) {
+            northbound.getAllLinks().each { assert it.state != IslChangeType.FAILED }
+        }
+        database.resetCosts()
+    }
```

Comment:
Please note that if the first flow is about to 'release' that bandwidth after swap and occupy it on some other ISL - then the swap should succeed. In other words: restrictions should calculate as if the flows are _already_ swapped, not _before_ the swap. Does it make sense?

Reply:
Fixed.